pkg/plugins/common/kustomize/v1/init.go

```diff
@@ -94,6 +108,30 @@ func (p *initSubcommand) InjectConfig(c config.Config) error {
 	return nil
 }
 
+func (p *initSubcommand) PreScaffold(machinery.Filesystem) error {
+	arch := runtime.GOARCH
+	arch = strings.Replace(arch, "x86_64", "amd64", -1)
+	localPlatform := fmt.Sprintf("%s/%s", strings.TrimSpace(runtime.GOOS), strings.TrimSpace(arch))
+	found := false
+	for _, value := range supportedArchs {
+		if value == localPlatform {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		log.Warnf("the architecture of this environment (%s) is not suppported by kustomize v3 (%s) which is "+
```

Comment:
Might be nicer to say platform rather than architecture here since we are mentioning which platforms are supported (i.e. linux/amd64 rather than just amd64)
```suggestion
		log.Warnf("the platform of this environment (%s) is not suppported by kustomize v3 (%s) which is "+
```